include/eve/detail/function/simd/arm/neon/slide_right.hpp

```diff
@@ -12,52 +12,60 @@
 
 namespace eve::detail
 {
-  template<real_scalar_value T, typename N, std::ptrdiff_t Shift>
-  EVE_FORCEINLINE wide<T,N> slide_right_( EVE_SUPPORTS(neon128_)
-                                            , wide<T,N> v, index_t<Shift>
-                                            ) noexcept
-  requires(Shift<=N::value) && arm_abi<abi_t<T, N>>
+
+  template <typename T, typename N, std::ptrdiff_t Shift>
+  EVE_FORCEINLINE wide<T, N> slide_right_arm_extract( wide<T,N> x, wide<T,N> y, index_t<Shift> )
   {
-    using that_t  = wide<T,N>;
+    constexpr auto c = categorize<wide<T, N>>();
 
-    if constexpr(Shift == N::value)
-    {
-      return that_t{0};
-    }
-    else if constexpr(Shift == 0)
-    {
-      return v;
-    }
+    // This works for smaller than 8 byte wides on a slide right with 0.
+    // I don't understand why but it does, I tested.
+    // The previous comment, that I didn't understand was:
+    //   The actual shift uses the expected cardinal so it has enough
+    //   to push through the unused space.
```

Comment:
Can you explain to me why in the world does the extract works for 2 smaller wides on arm?